docs/kb/integrations/customize-semgrep-precommit.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Understand how to customize Semgrep's behavior when using it with pre-commit.
+tags:
+  - Extensions
+---
+
+# Customize Semgrep in pre-commit
+
+Semgrep offers two standard pre-commit hooks [outlined here](/docs/extensions/overview/#pre-commit).
+
+* The `semgrep` hook is optimized to run with a provided set of rules, either local rules or rules from the Semgrep Registry.
+* The `semgrep-ci` hook is optimized to run with the rules from your organization in the Semgrep AppSec Platform, without sending results to the platform, since results from pre-commit are temporary and not linked to the final code in the repository.
+
+You can also create your own Semgrep hooks for pre-commit if you have particular needs or preferences, such as if you want users to see the output of the Semgrep scan even if it passed, or if you want to limit the scan to a particular Semgrep product.
+
+## Show the scan output
+
+By default, pre-commit does not show the scan output if the hook passes. If you're using the `semgrep-ci` hook, but want your developers to see the output even if non-blocking findings are found, you can set the `verbose` option to print the output. To limit output to findings only (no scan information or diagnostics), redirect `stderr` to `/dev/null`, or use `--quiet`.
+
+```yaml
+repos:
+- repo: https://github.com/semgrep/pre-commit
+  rev: 'v1.101.0'
+  hooks:
+    - id: semgrep-verbose
+      entry: semgrep
+      args: ["ci", "--dry-run", "--baseline-commit", "HEAD" "2>/dev/null"]
+      verbose: true
+      pass_filenames: false
+```
+
+## Limit the scan to a particular product
+
+Semgrep Secrets is an ideal product to run before commit, since it can help prevent secrets from ever making it into the Git history, even locally. To run only Secrets in pre-commit, add the product flag to the `args`:
+
+```yaml
+- repo: https://github.com/semgrep/pre-commit
+  rev: 'v1.101.0'
+  hooks:
+    - id:  semgrep-secrets
+      pass_filenames: false
+      args: ["ci", "--dry-run", "--baseline-commit", "HEAD", "--secrets"]
+```
+
+## Customization tips
+
+### Useful arguments
+
+The provided hooks include useful arguments for Semgrep that you may want to include in your hooks as well.
+
+For example, for the `semgrep` hook that runs with a provided `--config`, the arguments also include:
```

Comment:
we probably want --pro as a useful argument too

Reply:
This text is meant to cover "hey some useful arguments are used already, I'm going to explain them", not to introduce other arguments. But I can add an example or a note about `--pro` separately.

Reply:
gotcha!